### ISSUE TITLE
Error massage during making csv file with CompetingPhasesAnalyzer

### ISSUE THREAD

**Issue by kantaimperial** [SMTG-UCL/doped#16]
I got this error message when using CompetingPhasesAnalyzer to generate a csv file, although the file seems to be generated successfully. In addition, it says "parsing 33 vaspruns,", but only 28 phases are listed in the csv file.

```python
cpa.from_vaspruns(path="./PhadeDiagram/", folder="", csv_fname="STO_competing_Phase.csv"

Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrAl4O7_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/home, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr2Ti6O13_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Al_EaH=0.011, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Ti2O_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrAl2O4_EaH=0.012, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr_EaH=0.013, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/TiO2_EaH=0.0081, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/.DS_Store, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr_EaH=0.0056, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrO2_EaH=0.0096, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/O2_EaH=0.0017, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Ti_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Ti_EaH=0.004, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrO2_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrAl2O4_EaH=0.006, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/TiO_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrTi11O20_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrAl2O4_EaH=0.019, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr_EaH=0.0064, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/O2_EaH=0.01, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrTiO3_EaH=0.00057, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Al_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/TiO2_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrTiO3_EaH=0.00088, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/TiO2_EaH=0.0064, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr_EaH=0.0079, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/O2_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrTi3Al8O19_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrTiO3_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/SrAl2O4_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Ti2O3_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/Sr3Ti2O7_EaH=0, proceed with caution
Can't find a vasprun.xml(.gz) file for PhadeDiagram/O2_EaH=0.012, proceed with caution
parsing 33 vaspruns, this may take a while

```

**Comment by brlec** [comment]
Hi! I don't know why it's telling you the vaspruns aren't there (when they clearly are?) so will look into that, but the reason why it only lists 28 phases in the csv file is that you have multiple elemental phases for each of the elements - doped will only keep the lowest energy elemental phase. If you can double check your csv, the only ones missing should be the extra Sr, Ti, Al and O phases and all the other binaries, ternaries and quaternaries should be there

Also if you used doped to set up your competing phase analysis only the gaseous O2 molecule should have been generated for the oxygen elemental phase so also not sure what happened there 😬 

Hope this explains it

**Comment by kavanase** [comment]
Those false `vasprun.xml` warnings are now fixed in [27949a5](https://github.com/SMTG-UCL/doped/commit/27949a5be0cd477aabf5403f5d2fe09416bca2b6)

**Comment by kavanase** [comment]
Most of these issues should now be fixed with 27949a5be0cd477aabf5403f5d2fe09416bca2b6and 92c5f51d3ee1fe2883ae356ff01a0b6e73f18c59, though there are still some other updates to come with having the option to generate the 'bordering competing phases' vs full phase diagram with the new `CompetingPhases` classes. 

One thing that doesn't make sense yet though, in the `STO_competing_Phase.csv` file you sent me @kantaimperial, of the 28 phases included, there are multiple phases for both compounds and _elements_, meaning this doesn't explain the difference from "33 vaspruns" but only 28 phases in the notebook. Let's chat in the office to see your directory tree and data, to see where the discrepancy is coming from and if we can use some of this for internal testing 😃

**Comment by brlec** [comment]
if there are multiple elemental phases then clearly the code wasn't working as intended, mea culpa 🙇 i've not come across this when using doped to parse competing phases tho

**Comment by kavanase** [comment]
All fixed now in 2e44f3be24034cacbaa080cf3fb69546fa9ea155, thanks again for raising @kantaimperial!
